apheleia-formatters.el

```diff
@@ -44,6 +44,7 @@
     (cmake-format . ("cmake-format" "-"))
     (crystal-tool-format . ("crystal" "tool" "format" "-"))
     (dart-format . ("dart" "format"))
+    (dprint . ("dprint" "fmt" "--stdin" (file-name-nondirectory filepath)))
```

Comment:
My elisp is rusty and this fails:
```
Debugger entered--Lisp error: (void-variable filepath)
  (file-name-nondirectory filepath)
```

We need to pass a file basename only (or extension only), dprint treats full paths specially; in particular they must exist which causes problems e.g. with the test suite at the moment, and I suspect it might in other scenarios, too.

Reply:
It's not your fault. You're thinking that `filepath` is an ordinary variable, which would make sense. But it's actually a special symbol that is replaced before the command line is evaluated. The replacement only happens at the top level.

I think it would be reasonable to bind `filepath` and related symbols when evaluating non-string forms as mentioned in the docstring in `apheleia-formatters`. Then what you did would work.

By the way, I think you found a bug in the test framework. I think whatever `filepath` expands to _should_ exist during test execution. There might have been an oversight where that is not the case currently.

Reply:
I looked into this. The input file buffer has its buffer name set to an absolute filepath (unusual):

https://github.com/scop/apheleia/blob/cf05ccf000c66478151cc9d5897ac8597f0b68fc/test/formatters/apheleia-ft.el#L295

The `filepath` symbol is replaced here:

https://github.com/scop/apheleia/blob/cf05ccf000c66478151cc9d5897ac8597f0b68fc/apheleia-formatters.el#L894-L898

Because the buffer name is not supposed to be critically important it is sanitized:

https://github.com/scop/apheleia/blob/cf05ccf000c66478151cc9d5897ac8597f0b68fc/apheleia-formatters.el#L810-L817

That explains the unusual path name showing up. The bug is that the buffer name is used at all. Obviously, a buffer name does not a `filepath` make. The bug is that the test framework is not setting `buffer-file-name` in the test buffer, which is causing Apheleia to fall back to a non-existent `filepath` based on the buffer name.

I will fix this.